File: 1663-smallest-string-with-a-given-numeric-value/1663-smallest-string-with-a-given-numeric-value.js
/**
 * @param {number} n
 * @param {number} k
 * @return {string}
 */
var getSmallestString = function(n, k) {
  k -= n;
  let [curr, ans] = [0, ''];
  
  while (n--) {
    curr = Math.min(25, k);
    ans = String.fromCharCode(curr + 97) + ans;
    k -= curr;
  }
  
  return ans;
};

File: 1663-smallest-string-with-a-given-numeric-value/1663-smallest-string-with-a-given-numeric-value.py
class Solution:
    def getSmallestString(self, n: int, k: int) -> str:
        k -= n
        curr = 0
        ans = ''
        
        while n:
            curr = min(25, k)
            ans = chr(curr + 97) + ans
            k -= curr
            n -= 1
        return ans